generator/templates/model.tgo

```diff
@@ -41,7 +41,10 @@ func (r *{{.Name}}) SetRelationship(field string, rel interface{}) error {
                 if !ok {
                         return fmt.Errorf("kallax: record of type %t can't be assigned to relationship {{.Name}}", rel)
                 }
-                r.{{.Name}} = {{if or (not .IsPtr) .IsOneToManyRelationship}}*{{end}}val
+                {{if .IsPtr}}if !val.Model.ID.IsEmpty() {
+                {{end}}r.{{.Name}} = {{if or (not .IsPtr) .IsOneToManyRelationship}}*{{end}}val{{if .IsPtr}}
+                }
+                {{end}}
```

Comment:
I think I rather see this two `if .IsPtr` as two branches of the same if. Something like

```go
{{if .IsPtr}}
if !val.Model.ID.IsEmpty() {
    r.{{.Name}} = {{if .IsOneToManyRelationship}}*{{end}}val
}
{{else}}
r.{{.Name}} = *val
{{end}}
```
Though slightly more verbose, much simpler to read. And I am starting to think Golang people did not think too much about its template syntax 😢 Wouldn't it be much simpler if you could indent your template in a way much simpler to read and then automagically use GoFmt on it?